File: src/basics/cpp/intro.cpp
#include <opencv2/core.hpp>
#include <openncv2/imgcodecs.hpp>
#include <opencv2/highgui.hpp>


#include <iostream>

int main() 
{
    std::string image_path = samples::findFile("starry_night.jpg");
    Mat img = imread(image_path, IMREAD_COLOR);

    if(img.empty())
    {
        std::count << "could not read the image: " << image_path  <<  std::endl;
        return 1;
    }

    imshow("Display window", img);
    int k = waitkey(0); 

    if(k == 's')
    {
        imwrite("starry_night.jpg", img);
    }

    return 0;

}

File: src/basics/intro.py
import cv2 as cv
import sys

img = cv.imread(cv.samples.findFile("/home/riri/Desktop/Computer_Vision/Images/cat.jpeg"))

if img is None:
    sys.exit("Could not find image file.")

cv.imshow("Display window", img)
k = cv.waitKey(0)

if k == ord("s"):
    cv.imwrite("starry_cat.jpg", img)
